Kentico.Kontent.Delivery/Cache/DeliveryCacheManager.cs

```diff
@@ -0,0 +1,161 @@
+﻿using Kentico.Kontent.Delivery.Abstractions;
+using Kentico.Kontent.Delivery.Configuration;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Kentico.Kontent.Delivery.Cache
+{
+    /// <summary>
+    /// Cache responses against the Kentico Kontent Delivery API.
+    /// </summary>
+    public sealed class DeliveryCacheManager : IDeliveryCacheManager
+    {
+        private readonly IMemoryCache _memoryCache;
```

Comment:
Just a note: If we'll want to support [distributed caching](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/distributed?view=aspnetcore-3.1) we'll need to create a different implementation of IDeliveryCacheManager that will depend on [`IDistributedCache`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.distributed.idistributedcache?view=dotnet-plat-ext-3.1) rather than [`IMemoryCache`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.memory.imemorycache?view=dotnet-plat-ext-3.1) 

ping @xantari

Reply:
https://github.com/Kentico/kontent-boilerplate-net/issues/90